uint8 result
string result_message
uint8 ack
string ack_message
int64 data
